### ISSUE TITLE
UWP app - .Net Native tool chain issue with Release - assembliesToInclude???

### ISSUE THREAD

**Issue by faceoffers28** [luberda-molinet/FFImageLoading#301]
My app works just fine in Debug mode. When I Release the app, I can't get a couple of features to work. One of the features is FFImageLoading. I have read that you have to inject assemblies into the Xamarin.Forms.Forms.Init method, but I can't find the correctly assemblies to add. Here is what I have tried. Does anyone know the correct combination of assemblies to include to get this to work in Release mode? 

```
List<Assembly> assembliesToInclude = new List<Assembly>();

assembliesToInclude.Add(typeof(FFImageLoading.FFImage).GetTypeInfo().Assembly);
assembliesToInclude.Add(typeof(FFImageLoading.FFImageSourceBinding).GetTypeInfo().Assembly);
assembliesToInclude.Add(typeof(FFImageLoading.ImageService).GetTypeInfo().Assembly);
assembliesToInclude.Add(typeof(FFImageLoading.MiniLoggerWrapper).GetTypeInfo().Assembly);
assembliesToInclude.Add(typeof(FFImageLoading.PlatformPerformance).GetTypeInfo().Assembly);
assembliesToInclude.Add(typeof(FFImageLoading.TaskParameterExtensions).GetTypeInfo().Assembly);

assembliesToInclude.Add(typeof(FFImageLoading.Forms.CachedImage).GetTypeInfo().Assembly);
assembliesToInclude.Add(typeof(FFImageLoading.Forms.CachedImageEvents).GetTypeInfo().Assembly);
assembliesToInclude.Add(typeof(FFImageLoading.Forms.ImageSourceConverter).GetTypeInfo().Assembly);

assembliesToInclude.Add(typeof(FFImageLoading.Cache.CacheStream).GetTypeInfo().Assembly);

 Xamarin.Forms.Forms.Init(e, assembliesToInclude);

```

Here is how I have implemented.

```
<ffimageloading:CachedImage x:Name="imgPicked" VerticalOptions="CenterAndExpand" Aspect="AspectFill"
                        HeightRequest="300"
                        DownsampleToViewSize="true"
                        Source = "{Binding CurrentOffer.OfferImage}">
</ffimageloading:CachedImage>

```
Version info.
Install-Package Xamarin.FFImageLoading.Forms -Version 2.1.5

**Comment by faceoffers28** [comment]
OK, I finally got it to work by adding these assemblies.

```
assembliesToInclude.Add(typeof(FFImageLoading.Forms.WinUWP.CachedImageRenderer).GetTypeInfo().Assembly);
assembliesToInclude.Add(typeof(FFImageLoading.Forms.WinUWP.ImageSourceBinding).GetTypeInfo().Assembly);
```

Both of the packages that I was having trouble with have multiple DLLs in the Release bin (see image below).  Based on this revelation, I tried adding new assemblies based on the different DLLs. I'm still not sure what is the minimum amount of assemblies that I need to add in order for this to work or if these unused assemblies are adding extra overhead to the performance of the app. Does anyone know?

![aaa5](https://cloud.githubusercontent.com/assets/10619627/18145099/592807ae-6f8f-11e6-96a9-9601fd6dbd88.PNG)
